bijection-scrooge/src/main/scala/com/twitter/bijection/scrooge/ScroogeCodecs.scala

```diff
@@ -35,10 +35,19 @@ object BinaryScalaCodec {
 }
 
 class BinaryScalaCodec[T <: ThriftStruct](c: ThriftStructCodec[T])
-  extends ScalaCodec(new ThriftStructSerializer[T] {
+  extends Injection[T, Array[Byte]] {
+  import com.twitter.bijection.thrift.{ TArrayByteTransport, TArrayBinaryProtocol }
+
+  lazy val thriftStructSerializer = new ThriftStructSerializer[T] {
     override def codec = c
     val protocolFactory = new TBinaryProtocol.Factory
-  })
+  }
+
+  override def apply(item: T) = thriftStructSerializer.toBytes(item)
+  override def invert(bytes: Array[Byte]) = attempt(bytes){ bytes =>
```

Comment:
what about Macros.fastAttempt here to avoid the closure. :) Maybe no win, but it can't hurt.